tests/filecheck/transforms/riscv_cse.mlir

```diff
@@ -0,0 +1,51 @@
+// RUN: xdsl-opt -p riscv-cse %s | filecheck %s
+
+%0, %1, %2 = "test.op"() : () -> (!riscv.reg<>, !riscv.reg<>, !riscv.reg<>)
+
+%c0 = riscv.get_register : () -> !riscv.reg<zero>
+%c8 = riscv.li 8 : () -> !riscv.reg<>
+%c8_1 = riscv.li 8 : () -> !riscv.reg<>
+%c1 = riscv.li 1 : () -> !riscv.reg<>
+riscv_scf.for %arg3 : !riscv.reg<> = %c0 to %c8 step %c1 {
+    riscv_scf.for %arg4 : !riscv.reg<> = %c0 to %c8 step %c1 {
+        %3 = riscv.li 8 : () -> !riscv.reg<>
+        %4 = riscv.mul %3, %arg3 : (!riscv.reg<>, !riscv.reg<>) -> !riscv.reg<>
+        %5 = riscv.add %4, %arg5 : (!riscv.reg<>, !riscv.reg<>) -> !riscv.reg<>
+        riscv_scf.for %arg5 : !riscv.reg<> = %c0 to %c8 step %c1 {
+            %6 = riscv.li 8 : () -> !riscv.reg<>
+            %7 = riscv.mul %3, %arg3 : (!riscv.reg<>, !riscv.reg<>) -> !riscv.reg<>
+            %8 = riscv.add %4, %arg5 : (!riscv.reg<>, !riscv.reg<>) -> !riscv.reg<>
+            %9 = riscv.li 8 : () -> !riscv.reg<>
```

Comment:
I'm not sure I understand what happens with this specific `li` operation.
Does it get `cse`'d or `dce`'d?

Reply:
Good catch! I think it gets csed, since there's another `li 8` in the same block